### PR TITLE
Apply misc refactoring for a few function names

### PR DESCRIPTION
**Pull request by honggyukim**
This PR does misc refactoring for a few function names.
- check_libpthread() to has_dependency()
- load_module_symbol() to load_module_symbol_file()
- save_module_symbol() to save_module_symbol_file()

### CODE REVIEW

File: utils/symbol.h [hunk 1]
```diff
@@ -131,7 +131,7 @@ enum uftrace_trace_type {
 	TRACE_CYGPROF,
 };
 
-bool check_libpthread(const char *filename);
+bool check_library(const char *filename, const char *libname);
```
Comment (namhyung) [inline utils/symbol.h]: I'd suggest a more specific name - "has_dependency".

File: utils/symbol.c [hunk 2]
```diff
@@ -1771,7 +1771,7 @@ TEST_CASE(symbol_load_module) {
 
 	save_module_symbol(&stab, symfile, 0x400000);
```
Comment (namhyung) [inline utils/symbol.c]: Then I think you should rename `save_module_symbol_file()` as well. :)

Comment (honggyukim) [reply utils/symbol.c]: Done it. Thanks.

File: utils/symbol.h (hunk 1 above)
Comment (honggyukim) [reply utils/symbol.h]: Done.

**Comment by honggyukim** [comment]
I just rebased it again. Please take a look at it. Thanks.

**Pull request merged**